templates/workloads/partials/cf/subscribe.yml

```diff
@@ -0,0 +1,210 @@
+{{- if .Subscribe}}
+EventsKMSKey:
+  Type: AWS::KMS::Key
+  Properties:
+    KeyPolicy:
+      Version: '2012-10-17'
+      Statement:
+        - Sid: "Allow key use"
+          Effect: Allow
+          Principal:
+            AWS: !Sub 'arn:${AWS::Partition}:iam::${AWS::AccountId}:root'
+          Action:
+            - "kms:Create*"
+            - "kms:Describe*"
+            - "kms:Enable*"
+            - "kms:List*"
+            - "kms:Put*"
+            - "kms:Update*"
+            - "kms:Revoke*"
+            - "kms:Disable*"
+            - "kms:Get*"
+            - "kms:Delete*"
+            - "kms:ScheduleKeyDeletion"
+            - "kms:CancelKeyDeletion"
+            - "kms:Tag*"
+            - "kms:UntagResource"
+            - "kms:Encrypt"
+            - "kms:Decrypt"
+            - "kms:ReEncrypt*"
+            - "kms:GenerateDataKey*"
+          Resource: '*'
+        - Sid: "Allow SNS encryption"
```

Comment:
If I'm understanding correctly this will encrypt the messages for SQS, do we need to do something similar in `publish.yml` for the SNS topic?  
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sns-topic.html#cfn-sns-topic-kmsmasterkeyid

I can't tell if we have to use the same key for both resources or if we can create two independent keys 🤔 Would you be able to investigate that? 🙏

Reply:
It looks like only the SQS key needs to give SNS permissions, so this should be sufficient for allowing keys. The SNS Topic can have its own key, and we could set that to default if we want.